### ISSUE TITLE
Incorrect recommendation for `compact` for array unpacking statements on PHP 8.1

### ISSUE THREAD

**Issue by asgraf** [kalessil/phpinspectionsea#1906]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended)|
| Language level | PHP 8.1 |

Example code:
--
```php
$source_array = ['foo' => 1, 'bar' => 2, 'baz' => 3];
['foo' => $foo, 'bar' => $bar] = $source_array;
```

Current behavior:
--
`Use compact(...) instead` inspection appears
Applying it breaks code by changing it to:
```php
$source_array = ['foo' => 1, 'bar' => 2, 'baz' => 3];
compact('foo', 'bar') = $source_array;
```
Expected behavior:
--
`Use compact(...) instead` inspection should not appear for array unpacking statements

**Comment by ea-inspections-team** [comment]
Thank you for reporting @asgraf!
The false positive has been resolved.
